### PR TITLE
Enable mesh filter

### PR DESCRIPTION
**Pull request by JafarAbdi**
### Description

This's required for this [PR](https://github.com/ros-planning/moveit_tutorials/pull/558)

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Review by rhaschke — COMMENTED** [review]
Generally approved. See minor remarks below.
For the future, please be more verbose in your initial PR comment. I had to guess that the self_filter nodelet sources were lying around, but not (yet) actually build into a library.

**Comment by JafarAbdi** [comment]
> For the future, please be more verbose in your initial PR comment. I had to guess that the self_filter nodelet sources were lying around, but not (yet) actually build into a library.

I'm so sorry for not adding more info when creating the PR

**Review by rhaschke — APPROVED** [review]

File: moveit_ros/perception/mesh_filter/CMakeLists.txt [hunk 1]
```diff
@@ -24,8 +32,8 @@ if (CATKIN_ENABLE_TESTING)
   endif()
 endif()
 
-install(TARGETS ${MOVEIT_LIB_NAME}
+install(TARGETS ${MOVEIT_LIB_NAME} moveit_depth_self_filter
         LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
         ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
         RUNTIME DESTINATION ${CATKIN_GLOBAL_BIN_DESTINATION})
-install(DIRECTORY include/ DESTINATION ${CATKIN_GLOBAL_INCLUDE_DESTINATION})
+install(DIRECTORY include/ DESTINATION ${CATKIN_GLOBAL_INCLUDE_DESTINATION})
```
Comment (tylerjw) [inline moveit_ros/perception/mesh_filter/CMakeLists.txt:39]: [nit] newline at eof

Comment (tylerjw) [inline moveit_ros/perception/mesh_filter/CMakeLists.txt:39]: [nit] newline at eof

File: moveit_ros/perception/mesh_filter/src/depth_self_filter_nodelet.cpp [hunk 2]
```diff
@@ -174,25 +176,28 @@ void mesh_filter::DepthSelfFiltering::filter(const sensor_msgs::ImageConstPtr& d
...
+      }
     }
   }
 }
 
 // Handles (un)subscribing when clients (un)subscribe
 void mesh_filter::DepthSelfFiltering::connectCb()
 {
-  lock_guard<mutex> lock(connect_mutex_);
+  boost::lock_guard<boost::mutex> lock(connect_mutex_);
```
Comment (tylerjw) [inline moveit_ros/perception/mesh_filter/src/depth_self_filter_nodelet.cpp:200]: Is there a reason you are using a boost mutex and not the one from the stl?  Is it needed somewhere that takes only the boost type?

File: moveit_ros/perception/mesh_filter/src/transform_provider.cpp [hunk 3]
```diff
@@ -61,7 +55,7 @@ TransformProvider::~TransformProvider()
 void TransformProvider::start()
 {
   stop_ = false;
-  thread_ = thread(&TransformProvider::run, this);
+  thread_ = boost::thread(&TransformProvider::run, this);
```
Comment (tylerjw) [inline moveit_ros/perception/mesh_filter/src/transform_provider.cpp:58]: Also, what about using the stl thread?  Is this being used with something external that only takes the boost thread?

File: moveit_ros/perception/moveit_depth_self_filter.xml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<library path="lib/libmoveit_depth_self_filter">
+    <class name="mesh_filter/DepthSelfFiltering" type="mesh_filter::DepthSelfFiltering" base_class_type="nodelet::Nodelet">
+        <description>
+            TODO
+        </description>
+    </class>
+</library>
```
Comment (tylerjw) [inline moveit_ros/perception/moveit_depth_self_filter.xml:7]: [nit] newline at eof

**Review by tylerjw — COMMENTED** [review]
Some minor nits and questions.  I've tested this a bit with the update to the tutorial from Abi.  I haven't approved the tutorial yet because rviz crashes sometimes with it and I haven't figured out why.

**Review by henningkayser — APPROVED** [review]
Any reason this hasn't been merged yet?

**Pull request merged**